language/tools/move-cli/tests/cli_tests.rs

```diff
@@ -39,17 +41,193 @@ fn run_metatest() {
+    assert!(output.contains("Error: Invalid Api token"), "{}", output);
+
+    clean_up(&absolute_package_path);
+}
+
+#[test]
+fn upload_package_to_movey_prints_hardcoded_error_message_if_server_respond_5xx() {
+    let package_path = format!("{}/valid_package3", UPLOAD_PACKAGE_PATH);
+    init_git(&package_path, true);
+    let server = MockServer::start();
+    let server_mock = init_server_mock(&server, 500, Some("Invalid Api token"));
+    init_stub_registry_file(&package_path, &server.base_url());
+    let relative_package_path = PathBuf::from(&package_path);
+    let absolute_package_path =
+        path_to_string(&relative_package_path.canonicalize().unwrap()).unwrap();
+
+    let cli_exe = env!("CARGO_BIN_EXE_move");
+    let output = Command::new(cli_exe)
+        .env("MOVE_HOME", &absolute_package_path)
+        .current_dir(&absolute_package_path)
+        .args(["movey-upload"])
+        .output()
+        .unwrap();
+
+    server_mock.assert();
+    assert!(!output.status.success());
+    let output = String::from_utf8_lossy(output.stderr.as_slice()).to_string();
+    assert!(
+        output.contains("Error: An unexpected error occurred. Please try again later"),
+        "{}",
+        output
+    );
+
+    clean_up(&absolute_package_path);
+}
+
+#[test]
+fn upload_package_to_movey_with_no_remote_should_panic() {
+    let package_path = format!("{}/no_git_remote_package", UPLOAD_PACKAGE_PATH);
+    init_git(&package_path, false);
+
+    let cli_exe = env!("CARGO_BIN_EXE_move");
+    let output = Command::new(cli_exe)
+        .current_dir(&package_path)
+        .args(["movey-upload"])
+        .output()
+        .unwrap();
+
+    assert!(!output.status.success());
+    let error = String::from_utf8_lossy(output.stderr.as_slice()).to_string();
+    assert!(error.contains("invalid git repository"));
+
+    clean_up(&package_path);
+}
+
+// is_valid == true: all git commands are run
+// is_valid == false: missing git remote add command
+fn init_git(package_path: &str, is_valid: bool) {
+    Command::new("git")
+        .current_dir(package_path)
+        .args(&["init"])
+        .output()
+        .unwrap();
+    Command::new("git")
+        .current_dir(package_path)
+        .args(&["add", "."])
+        .output()
+        .unwrap();
+    if is_valid {
+        Command::new("git")
+            .current_dir(package_path)
+            .args(&[
+                "remote",
+                "add",
+                "test-origin",
+                "git@github.com:move-language/move.git",
+            ])
+            .output()
+            .unwrap();
+    }
+}
+
+fn clean_up(package_path: &str) {
+    let _ = fs::remove_file(format!("{}/simple_file", package_path));
+    fs::remove_dir_all(format!("{}/.git", package_path)).unwrap();
+    let credential_path = format!("{}{}", package_path, MOVEY_CREDENTIAL_PATH);
+    let _ = fs::remove_file(&credential_path);
+}
+
+fn init_stub_registry_file(package_path: &str, base_url: &str) {
```

Comment:
Short documentation comments for this function (and others, in particular `init_server_mock`) would be very useful.